### ISSUE TITLE
MacOsX Build failed

### ISSUE THREAD

**Issue by jvernet** [joncampbell123/dosbox-x#1392]
Hi,

trying to build under MacOsX 10.13.6, using the script ./dosbox-x/build-scripts/macosx/make-macosx-build.sh 

Failed with message:

 U```
ndefined symbols for architecture x86_64:
  "DOSBoxMenu::unassigned_item_handle", referenced from:
      DOSBoxMenu::displaylist_clear(DOSBoxMenu::displaylist&) in libgui.a(menu.o)
      ConstructMenu() in libgui.a(menu.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by jvernet** [comment]
May be here:

> ar cru libgui.a sdlmain_linux.o sdlmain.o sdl_mapper.o render.o render_scalers.o midi.o sdl_gui.o menu.o bitop.o ptrop.o zipcrc.o zipfile.o  menu_osx.o 
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libgui.a(sdlmain_linux.o) has no symbols
> ranlib libgui.a
> /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ranlib: file: libgui.a(sdlmain_linux.o) has no symbols
> [jeromevernet@macbook gui]$ ls

**Comment by joncampbell123** [comment]
There may have been changes to the code that break the Mac OS X build. I'll pull out my Macbook to check.

**Comment by Allofich** [comment]
I narrowed it down to
https://github.com/joncampbell123/dosbox-x/commit/06f77027e81446c95e60e2b9e5f80e3a2008ec09

That was to silence the cppCheck warning:
`Consider using std::fill algorithm instead of a raw loop.`

**Comment by Allofich** [comment]
I wonder why @joncampbell123 has been able to release macosx releases, though. The commit that causes the issue is from back in November, and there have been two releases since then.

I used Travis-CI and the `build-macosx` script in the root directory (not ./dosbox-x/build-scripts/macosx/make-macosx-build.sh like the bug reporter) for testing, but the error was the same as what's reported in this issue, and reverting the commit above allowed a successful build.

**Comment by joncampbell123** [comment]
I just did a build last night with the SDL1 and SDL2 build scripts with no issues.

**Comment by Allofich** [comment]
Hmm. Well, I'm taking down the pull request. It's a simple one-line change that can be done at any time, including by the bug reporter if they want.

**Comment by Allofich** [comment]
I also got it compiling (in Travis-CI) by moving `DOSBoxMenu::unassigned_item_handle` out from the call to `std::fill` by assigning it to a variable first and then passing the variable in to the `std::fill` call, so I reopened the PR since this also keeps the Cppcheck fix.

I'm interested if it will also fix the bug reporter's problem.

**Comment by Allofich** [comment]
The PR was merged, so @jvernet can you check if you can successfully build with the latest code?

**Comment by jvernet** [comment]
Yes, it's working now. It seems that you are using your own copy of SDL1.2, wich have a bug with the mouse cursor (cursor shown as a yellow rectangle) on 10.13 and later. Grab latest SDL 1.2 sources to fixes that (can't make a capture of the screen, mouse cursor do not appear on the capture).
Keyboard mapping with SDL 1.2 is totally messed with my french keyboard

**Comment by joncampbell123** [comment]
DOSBox-X uses it's own heavily modified version of SDL 1.2. It will however use whatever version of SDL 2.x available.

**Comment by jvernet** [comment]
Apply this patch to your SDL copy to get ride of the MacOsX cursor problem:

--- a/src/video/quartz/SDL_QuartzWM.m	Sat Aug 03 12:47:10 2019 +0300
+++ b/src/video/quartz/SDL_QuartzWM.m	Sat Aug 03 16:27:46 2019 -0400
@@ -55,7 +55,7 @@ /* copy data and mask, extending the mask to all black pixels because the inversion effect doesn't work with Cocoa's alpha-blended cursors */
for (i = 0; i < (w+7)/8*h; i++) {
-        planes[0][i] = data[i] ^ 0xFF;
+        planes[0][i] = ~data[i] & mask[i];
          planes[1][i] = mask[i] | data[i];
 }
